src/components/AppController.jsx

```diff
@@ -3,30 +3,90 @@ import BaseComponent from "lib/BaseComponent";
 import { setAppReady } from "lib/falcor/falcorUtils";
 import ReactTransitionGroup from 'react-addons-transition-group';
 import _ from "lodash";
-import { TransitionManager } from "lib/loader";
+import { TransitionManager } from "lib/loader"
+import { setContextForGlobalErrorFunctions, resetContextForGlobalErrorFunctions, setGlobalError } from 'lib/utilities';
 
 // Components
 import Navigation from "components/Navigation";
 import Footer from "components/Footer";
-import Loader from "components/Loader"
+import Loader from "components/Loader";
+import Alert from 'components/Alert';
 
 // Application CSS; applicationStyles alias,
 // CSS and SCSS loaders in webpack.config.js
 require('applicationStyles');
 
 export default class AppController extends BaseComponent {
+  constructor(props) {
+    super(props);
+    this.safeSetState({
+      error: null,
+      displayErrorMessage: false,
+    });
+  }
+
   componentDidMount() {
+    setContextForGlobalErrorFunctions.call(this);
+    window.onerror = (message, url, lineNo, columNo, error) => {
+      setGlobalError(error);
+    }
     super.componentDidMount();
     setAppReady();
   }
 
+  componentWillUnmount() {
+    super.componentWillUnmount();
+    resetContextForGlobalErrorFunctions();
+  }
+
   render() {
+    let lostInternetFlag = false;
```

Comment:
I think this is too much detail for the AppController, which should be kept as minimal as possible. You don't have to track errors on the state of AppController, it might make more sense to track it on the state of the Alert component itself. In any case, I would move all this logic here into Alert controller. The goal of AppController, should just be to give at a glance, a high level overview of the top level DOM and structure of the app.